c7n/resources/dynamodb.py

```diff
@@ -51,22 +53,52 @@ class resource_type(object):
         config_type = 'AWS::DynamoDB::Table'
 
     filter_registry = filters
-    retry = staticmethod(get_retry(('Throttled',)))
     permissions = ('dynamodb:ListTagsOfResource')
 
     def get_source(self, source_type):
         if source_type == 'describe':
             return DescribeTable(self)
         elif source_type == 'config':
-            return query.ConfigSource(self)
+            return ConfigTable(self)
         raise ValueError('invalid source %s' % source_type)
 
 
 register_universal_tags(Table.filter_registry, Table.action_registry, False)
 
 
+class ConfigTable(query.ConfigSource):
+
+    def load_resource(self, item):
+        resource = super(ConfigTable, self).load_resource(item)
+        resource['CreationDateTime'] = datetime.fromtimestamp(resource['CreationDateTime']/1000.0)
+        if 'LastUpdateToPayPerRequestDateTime' in resource['BillingModeSummary']:
+            resource['BillingModeSummary'][
+                'LastUpdateToPayPerRequestDateTime'] = datetime.fromtimestamp(
+                    resource['BillingModeSummary']['LastUpdateToPayPerRequestDateTime']/1000.0)
+
+        sse_info = resource.pop('Ssedescription', None)
+        if sse_info is None:
+            return resource
+        resource['SSEDescription'] = sse_info
+        for k, r in (('KmsmasterKeyArn', 'KMSMasterKeyArn'),
+                     ('Ssetype', 'SSEType')):
+            if k in sse_info:
+                sse_info[r] = sse_info.pop(k)
+        return resource
+
+
 class DescribeTable(query.DescribeSource):
 
+    def get_resources(self, *args, **kw):
+        # Dynamodb tables aren't taggable while pending creation, even
+        # attempting to fetch tags for a table will return not found errors.
+        # In order to resolve on several user reported issues  #3361, #3171, #3514
+        # When subscribing to create table events, sleep for 30s
+        if (self.manager.ctx.policy.execution_mode == 'cloudtrail' and
+            'CreateTable' in self.manager.data['mode']['events']):
+            time.sleep(30)
```

Comment:
rather than the sleep here could we employ a waiter (https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#waiters), haven' tested this yet but if the table isn't taggable until it 'exists' then it should work

Reply:
sounds good, done. underneath the hood it looks like exists waiter polls for state: 'ACTIVE'

Reply:
i'm a little unclear if the dynamic sizing is safe without weaving in the lambda context through, in the case of multiple tables, but given the actual scenario is create is not multi-table seems fine.